### PR TITLE
Fix the threshold, support `Weight` type

### PR DESCRIPTION
**Pull request by Megha-Dev-19**
image

### CODE REVIEW

**Comment by Megha-Dev-19** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@Megha-Dev-19 Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
You are true leader! Other sloths hot on your heels, keep moving!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/Megha-Dev-19?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/84&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/Megha-Dev-19?type=bot&pr=NEAR-DevHub/neardevhub-treasury-dashboard/84&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/Megha-Dev-19)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/Megha-Dev-19)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: instances/treasury-devdao.near/widget/lib/common.jsx
Comment (Tguntenaar) [inline instances/treasury-devdao.near/widget/lib/common.jsx:54]: What is the reason the required votes is different from the contract response?

**Review by Tguntenaar — APPROVED** [review]

**Comment by Tguntenaar** [comment]
Since it an asap bug if it is ready to merge you should merge it.

File: instances/treasury-devdao.near/widget/lib/common.jsx
Comment (Megha-Dev-19) [reply instances/treasury-devdao.near/widget/lib/common.jsx:54]: sorry I don't understand the question, different from contract meaning?
we don't get the exact number from contract, we only get the threshold (which can be a ratio or an absolute number), when it's a ratio we need to first calculate the no of people allowed to vote on that particular proposal type (i.e `approverAccounts`) and then get the absolute number, but in case it's already an absolute number we use that

Comment (frol) [inline instances/treasury-devdao.near/widget/lib/common.jsx:54]: I'd recommend you to consider implementing the exact logic from the contract here and use the same terminology. https://github.com/near-daos/sputnik-dao-contract/blob/3d568f9517a8c7a6510786d978bb25b180501841/sputnikdao2/src/policy.rs#L407-L425

When WeightVote is not involved, the requiredVotes is:

```
max(
  min(number_of_voters, quorum),
  min(
    floor(threshold.ratio * number_of_voters) + 1,
    number_of_voters
  )
)
```

I think currently, quorum rules are not taken into account at all.

**Review by frol — APPROVED** [review]
I left a couple of comments, but they are not blocker for the hotfix. Please, test, merge,  deploy, and verify that it works in production as expected.

File: instances/treasury-devdao.near/widget/lib/common.jsx
Comment (Tguntenaar) [reply instances/treasury-devdao.near/widget/lib/common.jsx:54]: I see, that makes sense. Thanks for the explanation.

**Comment by Tguntenaar** [comment]
@race-of-sloths score 3

**Pull request merged**
